### PR TITLE
init migration guide to 3.1.x

### PR DESCRIPTION
**Pull request by petersg83**

### CODE REVIEW

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  admin: {
+    auth: {
+      secret: env('ADMIN_JWT_SECRET', 'example-secret'),
+    },
+  },
+});
+```
+
+::: warning
+For security concerns, you must change `example-secret` by your own sophisticated secret.
```
Comment (derrickmehaffy) [inline docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:71]: May be a good idea to include a quick example of how to generate this? Say something like `node -e "console.log(require('crypto').randomBytes(64).toString('base64'));"`

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# Migration guide from 3.0.x to 3.1.x
+
+**Make sure your server is not running until the end of the migration**
+
+## Summary
+
+1. [Upgrading your dependencies](#_1-upgrading-your-dependencies)
+2. [Define the admin JWT Token](#_2-define-the-admin-jwt-token)
+3. [Migrate you custom admin panel plugins](#_3-migrate-your-custom-admin-panel-plugins)
```
Comment (derrickmehaffy) [inline docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:9]: `Migrate your`

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md (hunk 1 above)
Comment (derrickmehaffy) [reply docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:71]: Also the wording should probably be changed here "For security, you must change the `example-secret` to your own sophisticated and secure secret."

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  },
+});
+```
+
+::: warning
+For security concerns, you must change `example-secret` by your own sophisticated secret.
+:::
+
+:::tip NOTE
+All the administrator will be disconnected from the app and will need to log in again.
```
Comment (derrickmehaffy) [inline docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:75]: "All currently logged in administrators"

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+:::
+
+
+## 3. Migrate your custom admin panel plugins
+
+If you don't have custom plugins, you can jump to the next section.
+
+In  order to display your custom plugin link into the mail `LeftMenu` you need to update the plugin registration by adding `icon`, `name` and `menu` in the following file.
```
Comment (derrickmehaffy) [inline docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:83]: Looks like there is a double space between `In` and `order`

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md (hunk 1 above)
Comment (derrickmehaffy) [reply docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:71]: (Alternative for us Linux/Mac users for generating secrets that I use generally: `openssl rand 64 | base64`

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
+# Migration guide from 3.0.x to 3.1.x
+
+**Make sure your server is not running until the end of the migration**
+
+## Summary
```
Comment (alexandrebodin) [inline docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:5]: I think you can use the `[[toc]]` directive in vue-press to auto generate this

File: docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+In order to make the login work again you need to define the `secret` you want to use in `server.js`.
+
+**Example —** `config/server.js`
+
+```js
+module.exports = ({ env }) => ({
+  // ...
+  admin: {
+    auth: {
+      secret: env('ADMIN_JWT_SECRET', 'example-secret'),
```
Comment (alexandrebodin) [inline docs/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.md:66]: I am wondering if we should rather remove the default value and explain how to add the env var instead.

**Pull request merged**